src/UI/InputHandler.java

```diff
@@ -330,25 +325,20 @@ public boolean handleInput(InputAction action)
         break;
       case BACK:
         // Write keys file.
-        try
+        List<Pair<InputAction, String>> configItems = new ArrayList<Pair<InputAction,String>>();
+        for( InputAction key : bindingsByInputAction.keySet() )
         {
-          File keyFile = new File(KEYS_FILENAME);
-          FileWriter writer = new FileWriter(keyFile, false);
-          for( InputAction command: bindingsByInputAction.keySet() )
+          String val = "";
+          for( Integer code : bindingsByInputAction.get(key) )
           {
-            writer.write(command.toString());
-            for( Integer code : bindingsByInputAction.get(command) )
-            {
-              writer.write(" " + code);
-            }
-            writer.write('\n');
+            val += " " + code;
           }
-          writer.close();
-        }
-        catch( IOException ioe )
-        {
-          System.out.println("Error! Failed to save key bindings file!.\n  " + ioe.toString());
+          // substring to remove the first space
+          configItems.add(Pair.from(key, val.substring(1)));
         }
+        ConfigUtils.writeConfigItems(KEYS_FILENAME, configItems);
+        if( !ConfigUtils.writeConfigs(OTHER_INPUT_FILENAME, Arrays.asList(extraOptions)) )
+          System.out.println("Unable to write extra game input options to file.");
```

Comment:
Why check the return value for one of these but not the other?

Reply:
No reason, just lazy.